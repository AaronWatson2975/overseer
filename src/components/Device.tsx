import React, { Component } from "react";
import { AndroidDevice, Connection } from "../enums";
import { ListItem, ListItemIcon, ListItemText } from "@material-ui/core";
import {
  PhoneAndroid,
  TabletAndroid,
  DeveloperBoard,
  NetworkWifi,
  Usb
} from "@material-ui/icons";

interface Props {
  make: string;
  model: string;
  serial: string;
  duplicate: boolean;
  type: AndroidDevice;
  connection: Connection;
}

class Device extends Component<Props> {
  render() {
    let deviceIcon: JSX.Element = <PhoneAndroid />;
    let connectionIcon: JSX.Element = <Usb />;
    const deviceName = `${this.props.make} ${this.props.model} ${this.props
      .duplicate && this.props.serial}`;

    switch (this.props.type) {
      case AndroidDevice.Phone:
        deviceIcon = <PhoneAndroid />;
        break;

      case AndroidDevice.Tablet:
        deviceIcon = <TabletAndroid />;
        break;

      case AndroidDevice.DevBoard:
        deviceIcon = <DeveloperBoard />;
        break;
    }

    switch (this.props.connection) {
      case Connection.USB:
        connectionIcon = <Usb />;
        break;

      case Connection.Network:
        connectionIcon = <NetworkWifi />;
        break;
    }

    return (
      <ListItem button>
        <ListItemIcon>{deviceIcon}</ListItemIcon>
        <ListItemText primary={deviceName} />
      </ListItem>
    );
  }
}

export { Device };
