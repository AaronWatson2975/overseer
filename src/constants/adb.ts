export const DEFAULT_PORT = 5555;
