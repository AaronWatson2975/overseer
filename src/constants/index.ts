export * from "./mock.devices";
export * from "./adb";
