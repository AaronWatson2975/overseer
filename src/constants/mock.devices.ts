import { AndroidDevice, Connection } from "../enums";
import { IAndroidDevice } from "../interfaces";

/**
 * Data interface
 *
 * make - The make of the Androiddevice
 * model - The model of the Androiddevice
 * serial - The serial number of the Androiddevice
 * connection - The way the Androiddevice is connected
 * type - The type of Androiddevice(phone, tablet, etc.)
 * ip - The ip address of the Androiddevice
 * duplicate - Whether another Androiddevice of the same
 * make and model exists in the collection
 */

const mockDevices: IAndroidDevice[] = [
  {
    make: "Dragonboard",
    model: "820c",
    serial: "c0ffee11",
    connection: Connection.USB,
    type: AndroidDevice.DevBoard,
    ip: "127.0.0.1",
    duplicate: false,
    offline: false
  },
  {
    make: "Google",
    model: "Pixel 3a",
    serial: "a54bde3d",
    connection: Connection.USB,
    type: AndroidDevice.Phone,
    ip: "192.168.2.45",
    duplicate: true,
    offline: false
  },
  {
    make: "Google",
    model: "Pixel 3a",
    serial: "76cda2ea",
    connection: Connection.USB,
    type: AndroidDevice.Phone,
    ip: "192.168.2.48",
    duplicate: true,
    offline: false
  },
  {
    make: "Google",
    model: "Pixel C",
    serial: "1234abcd",
    connection: Connection.USB,
    type: AndroidDevice.Tablet,
    ip: "127.0.0.1",
    duplicate: false,
    offline: false
  },
  {
    make: "Inforce",
    model: "6640",
    serial: "c0ffee22",
    connection: Connection.Network,
    type: AndroidDevice.DevBoard,
    ip: "192.168.2.15",
    duplicate: false,
    offline: false
  },
  {
    make: "Samsung",
    model: "Galaxy S6",
    serial: "68cd4ef2",
    connection: Connection.USB,
    type: AndroidDevice.Phone,
    ip: "192.168.2.13",
    duplicate: false,
    offline: false
  },
  {
    make: "Samsung",
    model: "Galaxy S6",
    serial: "68cd4ef2",
    connection: Connection.Network,
    type: AndroidDevice.Phone,
    ip: "192.168.2.13",
    duplicate: false,
    offline: false
  },
  {
    make: "Raspberry Pi",
    model: "3b",
    serial: "c0ffee32",
    connection: Connection.Network,
    type: AndroidDevice.DevBoard,
    ip: "192.168.2.76",
    duplicate: false,
    offline: true
  }
];

export { mockDevices };
