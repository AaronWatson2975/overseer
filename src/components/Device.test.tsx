import { shallow } from "enzyme";
import React from "react";
import { Device } from "./Device";
import { mockDevices } from "../constants";

describe("doesn't crash", () => {
  it("renders without crashing", () => {
    shallow(<Device device={mockDevices[0]} />);
  });
});

test("basic snapshot test", () => {
  const wrapper = shallow(<Device device={mockDevices[0]} />);
  expect(wrapper).toMatchSnapshot();
});

test("duplicate test", () => {
  const wrapper = shallow(<Device device={mockDevices[1]} />);
  expect(wrapper).toMatchSnapshot();
});

test("offline test", () => {
  const wrapper = shallow(<Device device={mockDevices[7]} />);
  expect(wrapper).toMatchSnapshot();
});
