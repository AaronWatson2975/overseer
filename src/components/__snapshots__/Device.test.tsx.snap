// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot test 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  disabled={false}
>
  <WithStyles(ForwardRef(ListItemIcon))>
    <DeveloperBoardIcon />
  </WithStyles(ForwardRef(ListItemIcon))>
  <WithStyles(ForwardRef(ListItemText))
    primary="Dragonboard 820c "
  />
  <WithStyles(ForwardRef(ListItemIcon))>
    <UsbIcon />
  </WithStyles(ForwardRef(ListItemIcon))>
</WithStyles(ForwardRef(ListItem))>
`;

exports[`duplicate test 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  disabled={false}
>
  <WithStyles(ForwardRef(ListItemIcon))>
    <PhoneAndroidIcon />
  </WithStyles(ForwardRef(ListItemIcon))>
  <WithStyles(ForwardRef(ListItemText))
    primary="Google Pixel 3a (a54bde3d)"
  />
  <WithStyles(ForwardRef(ListItemIcon))>
    <UsbIcon />
  </WithStyles(ForwardRef(ListItemIcon))>
</WithStyles(ForwardRef(ListItem))>
`;

exports[`offline test 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  disabled={true}
>
  <WithStyles(ForwardRef(ListItemIcon))>
    <DeveloperBoardIcon />
  </WithStyles(ForwardRef(ListItemIcon))>
  <WithStyles(ForwardRef(ListItemText))
    primary="Raspberry Pi 3b   (Device is offline)"
  />
  <WithStyles(ForwardRef(ListItemIcon))>
    <NetworkWifiIcon />
  </WithStyles(ForwardRef(ListItemIcon))>
</WithStyles(ForwardRef(ListItem))>
`;
